### PR TITLE
allow Calculate fields in TdH case tiles

### PR DESCRIPTION
**Pull request by dannyroberts**
@NoahCarnahan 

I took the approach of using the xpath_function method (taking advantage of whatever the default is for the Enum / Calculate / etc.). (I also renamed `prop_name` to `xpath_function`, accordingly.)

In terms of tests:
- `app_case_tiles.json` shows the changes that ismaila was proposing (making the "header" and the "date" tile blocks Calculate fields; "header" with the name + age, "date" with a formatted date)
- `suite-case-tiles.xml` shows the total changes to the suite.xml output, which should now correspond to the changes is `app_case_tiles.json` the way we want them to.

### CODE REVIEW

**Comment by dannyroberts** [comment]
@ctsims a quick look from you would be good just to see if you see anything obviously wrong either with the plan or the output

**Comment by dannyroberts** [comment]
(It's hard for me to test this end-to-end locally because the TdH app uses mobile ucr and thus doesn't copy well.)

File: corehq/apps/app_manager/suite_xml/sections/details.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from __future__ import absolute_import
 from collections import namedtuple
 import os
-from xml.sax.saxutils import escape
+from xml.sax.saxutils import escape, quoteattr
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/sections/details.py]: F401 'quoteattr' imported but unused

**Comment by dannyroberts** [comment]
(by the way it's somewhat urgent since Ismaila needs to implement it after merging and deploying

**Comment by NoahCarnahan** [comment]
Looks good. Does your comment about copying the TDH app mean you haven't built an app with this change? It's a little annoying, but shouldn't take too long to create a case list with the necessary fields and a few forms...

**Pull request merged**